### PR TITLE
Add random query parameter to credentials request

### PR DESCRIPTION
**Pull request by perrygeo**

### CODE REVIEW

**Comment by scothis** [comment]
Please rename `rqp` to `_`

File: tests/test_upload.py [hunk 1]
```diff
@@ -12,6 +14,13 @@
 access_token = 'pk.{0}.test'.format(
     base64.b64encode(b'{"u":"testuser"}').decode('utf-8'))
 
+UUID1 = 4  # chosen by fair dice roll, guaranteed to be random
```
Comment (vincentsarago) [inline tests/test_upload.py]: `4` -> `00000000-0000-0000-0000-000000000000` to reflect the current format of `rqp` value ?

File: mapbox/services/uploads.py [hunk 2]
```diff
@@ -7,6 +9,16 @@
...
 
 
+@contextmanager
+def CacheBuster(session):
+    """Wraps a session with a randomized query parameter `_`
+    to discourage cached results.
+    """
+    session.params.update(_=str(uuid.uuid1()))
+    yield
+    session.params.pop('_')
```
Comment (sgillies) [inline mapbox/services/uploads.py]: The contextlib yield thing is weird but cool! `_` is a legit query string key and seems to indicate that it's not meaningful. @perrygeo I suppose there's a small risk that our API server could choke on a unexpected query. Other than that, everything looks good to me.

Comment (sgillies) [inline mapbox/services/uploads.py]: I missed the comment by @scothis. We're all good to go then.

File: tests/test_upload.py (hunk 1 above)
Comment (perrygeo) [inline tests/test_upload.py]: But then I can't make my [xkcd joke](https://imgs.xkcd.com/comics/random_number.png) :-P 

seriously though, it's probably good idea to make the mock uuid more realistic 👍

**Pull request merged**